File: brian2/codegen/runtime/java_rt/templates/threshold.java
////////////////////////////////////////////////////////////////////////////
//// MAIN CODE /////////////////////////////////////////////////////////////

{% macro main() %}
    // USE_SPECIFIERS { _num_neurons, not_refractory, lastspike, t }
    ////// SUPPORT CODE ///////
    {% for line in support_code_lines %}
    // {{line}}
    {% endfor %}

int run_{{codeobj_name}}() {
    //// MAIN CODE ////////////
    long _numspikes = 0;
    for (int _idx=0; _idx<_num_idx; _idx++) {
        const int _vectorisation_idx = _idx;
        {% for line in code_lines %}
        {{line}}
        {% endfor %}
        if(_cond) {
            _spikespace[_numspikes++] = _idx;
            not_refractory[_idx] = false;
            lastspike[_idx] = t;
        }
        _spikespace[_num_idx] = _numspikes;
        return _numspikes;
    }
}
{% endmacro %}

////////////////////////////////////////////////////////////////////////////
//// SUPPORT CODE //////////////////////////////////////////////////////////

{% macro support_code() %}
    {% for line in support_code_lines %}
    // {{line}}
    {% endfor %}
{% endmacro %}

File: brian2/codegen/runtime/java_rt/templates/threshold.rs
////////////////////////////////////////////////////////////////////////////
//// MAIN CODE /////////////////////////////////////////////////////////////

{% macro main() %}
//**** threshold.rs ****//
// USES_VARIABLES { not_refractory, lastspike, t, _spikespace }

void _run_{{codeobj_name}}(double t)
{

	//// MAIN CODE ////////////
	long _cpp_numspikes = 0;
	for(int _idx=0; _idx<_num_idx; _idx++)
	{
	    const int _vectorisation_idx = _idx;
		{% for line in code_lines %}
		{{line}}
		{% endfor %}
		if(_cond) {
			{{_spikespace}}[_cpp_numspikes++] = _idx;
			// We have to use the pointer names directly here: The condition
			// might contain references to not_refractory or lastspike and in
			// that case the names will refer to a single entry.
			_ptr{{_array_not_refractory}}[_idx] = false;
			_ptr{{_array_lastspike}}[_idx] = t;
		}
	}
	{{_spikespace}}[_num_idx] = _cpp_numspikes;
}
{% endmacro %}

